### ISSUE TITLE
[API Proposal]: `TimeSpan.ToString("c")` alternative that doesn't render days

### ISSUE THREAD

**Issue by Tyrrrz** [dotnet/runtime#88062]
### Background and motivation

A lot of tools that work with time strings (e.g. FFmpeg), as well as data formats (e.g. SRT), use the following format for rendering timestamps:

```
hh:mm:ss.fff
```

This is pretty similar to .NET's built-in `c` format, except that the latter has a component for days too:

```
[d.]hh:mm:ss.fff
```

It would be incredibly convenient to have a separate built-in format that works exactly like `c`, but instead of rendering days, it should just render the total number of hours, even if it's beyond 24 hours.

The current workaround is to do this:

```csharp
var value = TimeSpan.FromHours(27.123);

var str =
    Math.Floor(value.TotalHours).ToString("00", formatProvider) + ':' +
    value.Minutes.ToString("00", formatProvider) + ':' +
    value.Seconds.ToString("00", formatProvider) + '.' +
    value.Milliseconds.ToString("000", formatProvider);
```

This is pretty cumbersome.

### API Proposal

I suggest adding a new standard format, for example `ch`:

```csharp
var value = TimeSpan.FromHours(27.123);
var str = value.ToString("ch"); // 27:07:22.680
```

### API Usage

See above

**Comment by mattjohnsonpint** [comment]
I love this idea, but `ch` wouldn't work, because _standard_ format strings have to be a single character.  I propose capital `C` instead.  (There is already precedence for lower-case vs upper-case standard format strings.)  Or pick a new character completely.


Also, we'd have to solve a few related issues:
- We'd need a _custom_ format specifier for just the hours component, so that one could compose strings such as `"27:00"` without the seconds or fractional seconds components.  I suggest `H` and `HH`, which would work similarly to `h` and `hh` but, would allow exceeding 24 hours.
- We need to think through negative variations also.  There's currently no custom format specifier for the sign, and it's not included with either `h` or `hh`.  [The docs](https://learn.microsoft.com/dotnet/standard/base-types/custom-timespan-format-strings) currently call this out as a deficiency (in the second paragraph of the first blue "Important" box on the page).
- It's currently impossible to correctly _parse_ a string with more than 24 hours.  `TimeSpan.Parse("24:00")` fails outright.  `TimSpan.Parse("24:00:00")` thinks that should be 24 _days_.  `TimeSpan.ParseExact` with any parameters such as `h` or `hh` in the hours component fails because hours can't exceed 24.

**Comment by huoyaoyuan** [comment]
Note: a simpler workaround can be: `$"{(int)value.TotalHours}:{value:mm\\:ss\\.fff}"`.

**Comment by Tyrrrz** [comment]
> see [trac.ffmpeg.org/wiki/FFprobeTips#Duration](https://trac.ffmpeg.org/wiki/FFprobeTips#Duration). ffmpeg and srt don't use the exact same format. if there is something beyond srt using their format, or anything beyond ffmpeg using _their_ (different) format, then by all means go for it, otherwise it looks too niche to assign a formatting character.

Indeed, there are some subtle differences but they are generally negligible (for example, you can use both `,` and `.` in SRT to separate fractions). I think having a separate token for unbounded hours (e.g. `HHH`) would be nice to handle those things. But ultimately, I think even a standard format as described in the OP will be useful.

**Comment by crozone** [comment]
I think a better overall solution would be to go ahead and fix custom TimeSpan format strings. Currently custom TimeSpan format strings cannot even represent any of the standard Custom format strings which is a major shortcoming. Extending the custom format functionality would probably solve the need for any new standard format string specifiers.
